File: internal/models/device.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
	"gorm.io/gorm"
)

type Device struct {
	ID              uuid.UUID      `gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	TenantID        uuid.UUID      `gorm:"type:uuid;not null;index"`
	SerialNumber    string         `gorm:"type:varchar(255);not null;uniqueIndex"`
	Manufacturer    string         `gorm:"type:varchar(255);not null"`
	Model           string         `gorm:"type:varchar(255);not null"`
	FirmwareVersion string         `gorm:"type:varchar(100)"`
	DeviceType      string         `gorm:"type:varchar(100);not null"`
	Status          string         `gorm:"type:varchar(50);default:active"`
	PublicKey       string         `gorm:"type:text"`
	Metadata        datatypes.JSON `gorm:"type:jsonb"`
	CreatedAt       time.Time      `gorm:"autoCreateTime"`
	UpdatedAt       time.Time      `gorm:"autoUpdateTime"`
	DeletedAt       gorm.DeletedAt `gorm:"index"`
}

type DeviceFingerprint struct {
	ID              uuid.UUID      `gorm:"type:uuid;primaryKey;default:gen_random_uuid()"`
	DeviceID        uuid.UUID      `gorm:"type:uuid;not null;index"`
	FingerprintHash string         `gorm:"type:varchar(512);not null"`
	Embedding       datatypes.JSON `gorm:"type:jsonb"`
	CreatedAt       time.Time      `gorm:"autoCreateTime"`
}

File: frontend/src/types/device.ts
export interface DeviceFingerprint {
  id: string
  device_id: string
  serial_number: string
  manufacturer: string
  model: string
  firmware_version: string
  hardware_version: string
  public_key: string
  created_at: string
  updated_at: string
}

export interface Device {
  id: string
  tenant_id: string
  serial_number: string
  device_name: string
  status: DeviceStatus
  device_type: string
  fingerprint_id?: string
  fingerprint?: DeviceFingerprint
  last_seen?: string
  created_at: string
  updated_at: string
}

export type DeviceStatus = 'active' | 'inactive' | 'revoked' | 'compromised' | 'pending'

export interface DeviceStats {
  total: number
  active: number
  inactive: number
  revoked: number
  compromised: number
  pending: number
}
